src/ProjectOrigin.PedersenCommitment/Ristretto/Oracle.cs

```diff
@@ -0,0 +1,48 @@
+namespace ProjectOrigin.PedersenCommitment.Ristretto;
+using System.Text;
+
+class Oracle
+{
+
+    private List<byte[]> messages = new List<byte[]>();
```

Comment:
This will work. One could increase the efficiency by just storing the current state which corresponds essentially to the state of a hasher (see e.g. the implementation [here](https://github.com/Concordium/concordium-base/blob/main/rust-src/random_oracle/src/lib.rs)).

Reply:
While that would be preferable, it either requires doing the hashing in C# or keeping track of the state inside Rust. We could also just keep feeding it into the hasher at each call (hash to point, concat point bytes with new message, repeat), but I doubt that is faster than the extra allocations.

Reply:
The simplicity of having the entire oracle in Rust might be worth it though

Reply:
changed in cac3e273d1ee9c7ea4a30c541640a4a626bbc184

Reply:
> The simplicity of having the entire oracle in Rust might be worth it though

Indeed we will address this the 20. we also have a few colleagues who would love to wrestle with rust, and we will need this inhouse knowledge in order to make this project runnable in production.